src/client/compiler/parser/Parser.ts

```diff
@@ -11,12 +11,14 @@ type ASTNodes = ASTNode[];
 
 export class Parser {
 
-    static assignmentOperators = [TokenType.assignment, TokenType.plusAssignment, TokenType.minusAssignment, TokenType.multiplicationAssignment, TokenType.divisionAssignment, TokenType.moduloAssignment];
+    static assignmentOperators = [TokenType.assignment, TokenType.plusAssignment, TokenType.minusAssignment, TokenType.multiplicationAssignment, TokenType.divisionAssignment, TokenType.moduloAssignment, TokenType.ANDAssigment, TokenType.XORAssigment, TokenType.ORAssigment, TokenType.shiftLeftAssigment, TokenType.shiftRightAssigment, TokenType.shiftRightUnsignedAssigment];
 
     static operatorPrecedence: TokenType[][] = [Parser.assignmentOperators,
     [TokenType.ternaryOperator], [TokenType.colon],
-    [TokenType.or], [TokenType.and],
-    [TokenType.keywordInstanceof, TokenType.lower, TokenType.lowerOrEqual, TokenType.greater, TokenType.greaterOrEqual, TokenType.equal, TokenType.notEqual],
+    [TokenType.or], [TokenType.and], [TokenType.OR], [TokenType.XOR], [TokenType.AND],
+    [TokenType.equal, TokenType.notEqual],
+    [TokenType.keywordInstanceof, TokenType.lower, TokenType.lowerOrEqual, TokenType.greater, TokenType.greaterOrEqual],
+    [TokenType.shiftLeft, TokenType.shiftRight, TokenType.shiftRightUnsigned],
     [TokenType.plus, TokenType.minus], [TokenType.multiplication, TokenType.division, TokenType.modulo]
     ];

```

Comment:
I changed the order a bit to more closely relate to [java's operator precedence](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html)